lib/child_process.js

```diff
@@ -136,12 +136,17 @@ function normalizeExecArgs(command, options, callback) {
   }
 
   // Make a shallow copy so we don't clobber the user's options object.
-  options = Object.assign({}, options);
-  options.shell = typeof options.shell === 'string' ? options.shell : true;
+  const execOptions = Object.assign({}, options);
```

Comment:
Would using `Object.assign(Object.create(null), options)` remove the need for the `hasOwnProperty()` call?

Reply:
I think having tests to catch any future changes should be adequate, but it wouldn't be bad to update the existing comments. Right now the comments say that we don't want to clobber user input, but we could also mention that `Object.create(null)` is used so that only self properties are checked.

Reply:
Sounds good, will update 👍